### PR TITLE
Implements OrderItems resource for Paymentorder purchase, UpdateOrder, Capture & Reversal operations

### PR DESCRIPTION
**Pull request by shahriar-vdc**

### CODE REVIEW

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Helper/Factory/OrderItemFactory.php  5
- Helper/Factory/OrderItemsFactory.php  2
- Test/Unit/OrderItemFactoryTest.php  3
         

Complexity decreasing per file
==============================
+ Gateway/Command/Capture.php  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-checkout/pullRequest?prid=5835232&bid=19108447)

File: Helper/Factory/OrderItemFactory.php [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    /**
+     * @param OrderItemInterface $orderItem
+     * @return OrderItem
+     */
+    public function createByOrderItem(OrderItemInterface $orderItem)
+    {
+        $sku = $orderItem->getSku();
+        $name = $orderItem->getName();
+        $type = 'PRODUCT';
+        $itemClass = 'ProductGroup1';
```
Comment (asbjornu) [inline Helper/Factory/OrderItemFactory.php:56]: Please don't hard code `$itemClass` to `ProductGroup1`. If there's a "product category" or similar field in Magento you can use, please use that as long as you ensure it adheres to its specification:

>The classification of the order item. Can be used for assigning the order item to a specific product category, such as `MobilePhone`. Note that class cannot contain spaces and must follow the regex pattern `[\w]* (a-zA-Z0-9_)`. Swedbank Pay may use this field for statistics.

File: Helper/Factory/OrderItemFactory.php [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    /**
+     * @param QuoteItem $quoteItem
+     * @return OrderItem
+     */
+    public function createByQuoteItem(QuoteItem $quoteItem)
+    {
+        $sku = $quoteItem->getSku();
+        $name = $quoteItem->getName();
+        $type = 'PRODUCT';
+        $itemClass = 'ProductGroup1';
```
Comment (asbjornu) [inline Helper/Factory/OrderItemFactory.php:23]: Please don't hard code `$itemClass` to `ProductGroup1`. If there's a "product category" or similar field in Magento you can use, please use that as long as you ensure it adheres to its specification:

>The classification of the order item. Can be used for assigning the order item to a specific product category, such as `MobilePhone`. Note that class cannot contain spaces and must follow the regex pattern `[\w]* (a-zA-Z0-9_)`. Swedbank Pay may use this field for statistics.

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Helper/Factory/OrderItemsFactory.php  2
- Test/Unit/OrderItemFactoryTest.php  3
- Helper/Factory/OrderItemFactory.php  5
         

Complexity decreasing per file
==============================
+ Gateway/Command/Capture.php  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-checkout/pullRequest?prid=5835232&bid=19108447)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Helper/Factory/OrderItemFactory.php  5
- Test/Unit/OrderItemFactoryTest.php  3
- Helper/Factory/OrderItemsFactory.php  2
         

Complexity decreasing per file
==============================
+ Gateway/Command/Capture.php  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-checkout/pullRequest?prid=5835232&bid=19108447)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Test/Unit/OrderItemFactoryTest.php  3
- Test/Unit/QuoteRepositoryPluginTest.php  1
- Helper/Factory/OrderItemFactory.php  5
- Helper/Factory/OrderItemsFactory.php  2
         

Complexity decreasing per file
==============================
+ Gateway/Command/Capture.php  -7
         

Clones added
============
- Test/Unit/PaymentorderTest.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-checkout/pullRequest?prid=5835232&bid=19108447)

**Pull request merged**